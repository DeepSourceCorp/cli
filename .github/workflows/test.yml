name: Unit Tests

on:
  push:
    branches: [develop]
  pull_request:
    branches: [develop]

jobs:
  test:
    runs-on: ubuntu-latest
    env:
      NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
      ZEAL_SECRET: ${{secrets.GITHUB_TOKEN}}

    steps:
      - name: Checkout Bifrost
        uses: actions/checkout@v2

      - name: Setting Up Node
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com/
          scope: '@deepsourcelabs'
          cache: 'yarn'

      - name: Install packages
        run: yarn

      - name: Run Test
        run: yarn test --coverage=true --coverageReporters=cobertura

      - name: Install DeepSource CLI
        run: curl https://deepsource.io/cli | sh

      - name: Set DEEPSOURCE_DSN
        run: |
          ./bin/deepsource report --analyzer test-coverage --key javascript --value-file ./coverage/cobertura-coverage.xml
        env:
          DEEPSOURCE_DSN: ${{ secrets.BIFROST_DSN }}
