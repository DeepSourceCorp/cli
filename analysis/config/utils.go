package config

import (
	"log"
	"os"
	"path"
	"path/filepath"
	"strings"

	"github.com/bmatcuk/doublestar/v4"
)

/* Walks the `CODE_PATH` directory and returns all the files other than the ones present
 * in the .git folder for analysis in the form of a string array */
func readAllFiles(codePath string) ([]string, error) {
	fileCount := 0

	allFiles := make([]string, 0)

	err := filepath.Walk(codePath,
		func(path string, fileInfo os.FileInfo, err error) error {
			if err != nil {
				return err
			}
			fileCount++

			/* Check the following before appending to the list of files:
			 * - Should not be a directory
			 * - The walked file should not be present in .git folder
			 * TODO: Check why fileCount != 1 condition is used here */
			if !fileInfo.IsDir() || !strings.HasPrefix(path, filepath.Join(codePath, ".git")) && fileCount != 1 {
				allFiles = append(allFiles, path)
			}
			return nil
		})
	if err != nil {
		return allFiles, err
	}
	return allFiles, nil
}

// Returns the slice of files matching certain glob patterns
func (r *AnalysisRun) getMatchingFiles(patterns []string) ([]string, error) {
	matchedFiles := make([]string, 0)

	// Return all the files if no exclude_patterns are configured
	if len(patterns) == 0 {
		return matchedFiles, nil
	}

	for _, file := range r.AnalysisFiles {
		for i := range patterns {
			match, err := doublestar.Match(path.Join(r.CodePath, patterns[i]), file)
			if err != nil {
				log.Printf("Failed to match the file %s with pattern %s\n", file, patterns[i])
			}
			if match {
				matchedFiles = append(matchedFiles, file)
				break
			}
		}
	}
	return matchedFiles, nil
}

// Filters the analysis files and removes the files matching the exclude_patterns from them
// TODO: Improve the logic here
func (r *AnalysisRun) filterAnalysisFiles() {
	filteredFiles := []string{}
	for _, file := range r.AnalysisFiles {
		for _, excludedFile := range r.ExcludedFiles {
			if file == excludedFile {
				break
			}
			filteredFiles = append(filteredFiles, file)
		}
	}
	r.AnalysisFiles = filteredFiles
}
