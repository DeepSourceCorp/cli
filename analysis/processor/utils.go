package processor

import (
	"sort"

	"github.com/deepsourcelabs/cli/types"
)

// sortIssuesByFile sorts the issues in an alphabetical order according to the filenames
// where they got reported.
func (p *ProcessAnalysisResults) sortIssuesByFile() {
	sort.Slice(p.AnalysisResult.Issues, func(i, j int) bool {
		return p.AnalysisResult.Issues[i].Location.Path < p.AnalysisResult.Issues[j].Location.Path
	})
}

// Prepare a map with unique filenames as key and the issue range for each file as value
// This is done to ensure fewer loops when processing issues.
type IssueRange struct {
	Filename   string // The file which has the issues
	BeginIndex int    // Array index in report.Issues where the particular issue starts
	EndIndex   int    // Array index in report.Issues where the particular issue ends
}

// GenerateIssueRangeSlice generates an array containing the issue ranges with respect to files
// that helps us to go through them and map them to the files where they got reported instead
// of opening the file for each of them. The generated index looks like this:

// [{analyzer.go 0 0} {autofix_patch.go 1 1} {difftool.go 2 2} {patch.patch 3 5} {proc_skip_cq_test.go 6 6}]

// Here, the first field if filename and the second and third fields are the index range in which the issues reported
// in these files lie in the sorted AnalyzerReport slice.
func createIssueFileRange(report types.AnalysisResult) []IssueRange {
	fileCount := 0 // for 1 file, 0 based indexing
	issuesRange := []IssueRange{}
	prevFilename := report.Issues[0].Location.Path

	issueRange := IssueRange{
		BeginIndex: 0,
		EndIndex:   len(report.Issues) - 1,
		Filename:   prevFilename,
	}
	issuesRange = append(issuesRange, issueRange)

	// Iterating over the issues and creating an array containing issues with index data about
	// the files in which those issues are present
	for i := 1; i < len(report.Issues); i++ {
		issue := report.Issues[i]
		currentFilename := issue.Location.Path

		// TODO: Check when this condition is implied
		if issue.Location.Position.End.Line == -1 {
			issue.Location.Position.End.Line = issue.Location.Position.Begin.Line
		}

		if currentFilename != prevFilename {
			fileCount++

			issueRange = issuesRange[fileCount-1]
			issueRange.EndIndex = i - 1
			issuesRange[fileCount-1] = issueRange

			// Create for the new file
			issueRange = IssueRange{
				Filename:   currentFilename,
				BeginIndex: i,
				EndIndex:   len(report.Issues) - 1,
			}

			issuesRange = append(issuesRange, issueRange)
			prevFilename = currentFilename
		}
	}
	return issuesRange
}

// formatLSPResultsToDefault converts the LSP based analysis results into the default format supported by DeepSource.
func (p *ProcessAnalysisResults) formatLSPResultsToDefault() {
	p.AnalysisResult.IsPassed = p.Report.IsPassed
	p.AnalysisResult.Metrics = append(p.Report.Metrics, p.AnalysisResult.Metrics...)
	p.AnalysisResult.Errors = append(p.Report.Errors, p.AnalysisResult.Errors...)

	// Appending the issues to the default format of Analysis report
	for _, issue := range p.Report.Issues {
		analysisIssue := types.Issue{
			IssueCode: issue.Code,
			IssueText: issue.Message,
			Location: types.Location{
				Path: issue.RelatedInformation[0].Location.URI,
				Position: types.Position{
					Begin: types.Coordinate{
						Line:   issue.Range.Start.Line,
						Column: issue.Range.Start.Character,
					},
					End: types.Coordinate{
						Line:   issue.Range.End.Line,
						Column: issue.Range.End.Character,
					},
				},
			},
		}
		p.AnalysisResult.Issues = append(p.AnalysisResult.Issues, analysisIssue)
	}
}
