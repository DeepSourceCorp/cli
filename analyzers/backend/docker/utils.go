package docker

import (
	"bufio"
	"encoding/json"
	"errors"
	"fmt"
	"io"
	"strings"
)

type ErrorLine struct {
	Error       string      `json:"error"`
	ErrorDetail ErrorDetail `json:"errorDetail"`
}

type ErrorDetail struct {
	Message string `json:"message"`
}

type DockerBuildResponse struct {
	Stream string `json:"stream"`
}

func CheckBuildResponse(rd io.Reader, showAllLogs bool) error {
	var lastLine []byte
	count := 0
	var currentStream string

	scanner := bufio.NewScanner(rd)
	for scanner.Scan() {
		lastLine = scanner.Bytes()
		d := &DockerBuildResponse{}
		err := json.Unmarshal(lastLine, d)
		if err != nil {
			return err
		}
		if d.Stream == "" || d.Stream == "\n" || strings.Contains(d.Stream, "--->") || strings.TrimSuffix(d.Stream, "\n") == currentStream {
			continue
		}
		currentStream = strings.TrimSuffix(d.Stream, "\n")
		if showAllLogs {
			fmt.Println(currentStream)
		}
		count++
	}

	errLine := &ErrorLine{}
	json.Unmarshal([]byte(lastLine), errLine)
	if errLine.Error != "" {
		return errors.New(errLine.Error)
	}
	return scanner.Err()
}


