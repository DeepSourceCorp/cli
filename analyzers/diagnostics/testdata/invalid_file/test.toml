name = "2do-Checker"
