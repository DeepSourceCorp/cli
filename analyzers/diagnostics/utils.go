package diagnostics

import (
	"fmt"
	"os"
	"regexp"
	"strings"

	"github.com/morikuni/aec"
)

// fieldRegexp is a regular expression for checking the current line is a field or not.
var fieldRegexp = `\s*(?P<field>.+)\s*=\s*`

// readFileContent reads the file and returns its content.
func readFileContent(filename string) (string, error) {
	content, err := os.ReadFile(filename)
	if err != nil {
		return "", err
	}
	return string(content), nil
}

// prepareCodeFrame prepares a code frame using the file content. The code frame is meant to be displayed on the console while reporting diagnostics.
// NOTE: lineNum always starts from 0.
func prepareCodeFrame(lineNum int, lines []string) string {
	frame := ""

	if lineNum <= 1 {
		// Case 1: When the line is near the top of the file.
		// Generate a frame with the current and next line only.
		frame += aec.LightRedF.Apply(fmt.Sprintf("> %d | %s\n", lineNum+1, lines[lineNum]))
		frame += fmt.Sprintf("  %d | %s\n", lineNum+2, lines[lineNum+1])

	} else if lineNum >= (len(lines) - 1) {
		// Case 2: When the line is near the bottom of the file.
		// Generate a frame with the current line only.
		frame += aec.LightRedF.Apply(fmt.Sprintf("> %d | %s\n", lineNum, lines[lineNum-1]))
	} else {
		// Case 3: When the line is in between the top and the bottom.
		// Generate a frame with the the previous, current and the next line.
		frame += fmt.Sprintf("  %d | %s\n", lineNum, lines[lineNum-1])
		frame += aec.LightRedF.Apply(fmt.Sprintf("> %d | %s\n", lineNum+1, lines[lineNum]))
		frame += fmt.Sprintf("  %d | %s\n", lineNum+2, lines[lineNum+1])
	}

	return frame
}

// checkField checks if the line contains the field.
func checkField(line, field string) bool {
	// We use fieldRegexp for checking if the current line is a field or not.
	// If it's not a field, for example, comments for explanation, etc., then checkField should return false.
	exp := regexp.MustCompile(fieldRegexp)

	// Get index for the "field" named group.
	matches := exp.FindStringSubmatch(line)
	groupIndex := exp.SubexpIndex("field")

	// Capture match. We proceed only if the groupIndex isn't -1 (group not found), and if the number of matches aren't zero.
	// Once done, we populate match with the match found using the regular expression.
	match := ""
	if len(matches) != 0 && groupIndex != -1 {
		match = matches[groupIndex]
	}
	match = strings.TrimSpace(match)

	// Return true if the match contains the field.
	return strings.Contains(match, field)
}
