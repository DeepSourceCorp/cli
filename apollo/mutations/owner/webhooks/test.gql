mutation testWebhook($webhookId: ID!) {
  testWebhook(input: { webhookId: $webhookId }) {
    ok
  }
}
