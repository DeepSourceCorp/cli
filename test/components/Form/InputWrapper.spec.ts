import '@testing-library/jest-dom'
import { render } from '@testing-library/vue'

import InputWrapper from '~/components/Form/InputWrapper.vue'
import { cartesian, generateBooleanProps, generateStringProps } from '~/test/utils'

describe('[[ InputWrapper ]]', () => {
  test('renders `InputWrapper` with all prop options', () => {
    const baseProps = {
      label: 'Test label',
      description: 'Test description',
      inputId: 'test-input-id'
    }

    const removeYPaddingOptions = generateBooleanProps('removeYPadding', false)
    const cascadeInputOptions = generateBooleanProps('cascadeInput', false)

    const inputWidthOptions = generateStringProps('inputWidth', [
      'xx-small',
      'x-small',
      'small',
      'base',
      'wide'
    ])

    const customGridClassOptions = generateStringProps('customGridClass', [
      '',
      'grid-cols-1 md:grid-cols-4'
    ])

    cartesian(
      removeYPaddingOptions,
      cascadeInputOptions,
      inputWidthOptions,
      customGridClassOptions
    ).forEach((propCombination) => {
      const props = {
        ...baseProps,
        ...propCombination
      }
      const { html } = render(InputWrapper, {
        props
      })

      expect(html()).toMatchSnapshot(JSON.stringify(props))
    })
  })
})
