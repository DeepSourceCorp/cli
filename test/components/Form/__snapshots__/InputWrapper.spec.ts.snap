// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","cascadeInput":false,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-16 py-4\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","cascadeInput":false,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-16\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","cascadeInput":true,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-16 py-4\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","cascadeInput":true,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-16\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"base","cascadeInput":false,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-20 py-4\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"base","cascadeInput":false,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-20\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"base","cascadeInput":true,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-20 py-4\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"base","cascadeInput":true,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-20\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"small","cascadeInput":false,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-16 py-4\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"small","cascadeInput":false,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-16\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"small","cascadeInput":true,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-16 py-4\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"small","cascadeInput":true,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-16\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"wide","cascadeInput":false,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-24 py-4\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"wide","cascadeInput":false,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-24\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"wide","cascadeInput":true,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-24 py-4\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"wide","cascadeInput":true,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-24\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"x-small","cascadeInput":false,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-12 py-4\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"x-small","cascadeInput":false,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-12\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"x-small","cascadeInput":true,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-12 py-4\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"x-small","cascadeInput":true,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-12\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"xx-small","cascadeInput":false,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-8 py-4\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"xx-small","cascadeInput":false,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-8\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"xx-small","cascadeInput":true,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-8 py-4\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":"xx-small","cascadeInput":true,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-1 md:grid-cols-fr-8\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":null,"cascadeInput":false,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-fr-16 py-4\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":null,"cascadeInput":false,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-fr-16\\">
  <div class=\\"\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":null,"cascadeInput":true,"removeYPadding":false} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-fr-16 py-4\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;

exports[`[[ InputWrapper ]] renders \`InputWrapper\` with all prop options: {"label":"Test label","description":"Test description","inputId":"test-input-id","inputWidth":null,"cascadeInput":true,"removeYPadding":true} 1`] = `
"<div class=\\"grid gap-4 sm:gap-6 grid-cols-fr-16\\">
  <div class=\\"col-span-full md:col-auto\\"><label for=\\"test-input-id\\" class=\\"text-sm text-vanilla-100 mb-1 block\\"> Test label </label>
    <div class=\\"text-xs leading-5 text-vanilla-400\\">
      <p>Test description</p>
    </div>
  </div>
</div>"
`;
